operator/internal/controller/vectorized/cluster_controller_configuration.go

```diff
@@ -177,19 +179,31 @@ func (r *ClusterReconciler) applyPatchIfNeeded(
 	log := l.WithName("applyPatchIfNeeded")
 	errorWithContext := newErrorWithContext(redpandaCluster.Namespace, redpandaCluster.Name)
 
-	var invalidProperties []string
-	for i := range status {
-		invalidProperties = append(invalidProperties, status[i].Invalid...)
-		invalidProperties = append(invalidProperties, status[i].Unknown...)
-	}
-
-	patch := configuration.ThreeWayMerge(log, cfg.ClusterConfiguration, clusterConfig, lastAppliedConfiguration, invalidProperties, schema)
```

Comment:
Can we delete the `ThreeWayMerge` function now?

Reply:
Need to follow up by getting rid of the drift controller, which also uses it. I'll tidy up after that.